conda/core/repodata.py

```diff
@@ -74,6 +74,24 @@ def __call__(cls, channel):
 class SubdirData(object):
     _cache_ = {}
 
+    @staticmethod
+    def reverse_query_all(channels, subdirs, package_ref_spec):
+        from .index import check_whitelist  # TODO: fix in-line import
+        channel_urls = all_channel_urls(channels, subdirs=subdirs)
+        check_whitelist(channel_urls)
+        with ThreadLimitedThreadPoolExecutor() as executor:
+            futures = (executor.submit(
+                SubdirData(Channel(url)).reverse_query, package_ref_spec
+            ) for url in channel_urls)
+            return tuple(concat(future.result() for future in as_completed(futures)))
+
+    def reverse_query(self, package_ref_spec):
+        if not self._loaded:
+            self.load()
+        for prec in self._package_records:
+            if any(package_ref_spec.match(MatchSpec(drec)) for drec in prec.depends):
```

Comment:
You can't match a MatchSpec to a MatchSpec.  We haven't written in the capability to "merge", or determine overlaps, in e.g. version ranges.

Reply:
Do you want to use `prec.depends` or `prec.combined_depends`?

Reply:
> You can't match a MatchSpec to a MatchSpec.

Actually, that might be exactly the reason this got dropped from 4.4.  I think it's harder to implement than it seems on the surface.  It's going to take a lot of work to get `package_ref_spec.match(MatchSpec(drec))` right.

Reply:
Or... For now we could lock this down to just `package_ref`, and not `package_ref_or_match_spec`.  But then we'd have to change that API in the future if we do end up adding the matchspec compare/merge capability.